common/src/main/java/com/lightstep/tracer/shared/Options.java

```diff
@@ -214,6 +219,15 @@ public OptionsBuilder withTag(String key, Object value) {
         }
 
         /**
+         * Sets user-defined key-value pairs that should be sent in the HTTP header of all of the
+         * reports produced by this tracer.
+         */
+        public OptionsBuilder withCustomHeader(String key, String value) {
+            customHeaders.put(key, value);
+            return this;
+        }
```

Comment:
We might have anothre helper method that accepts `Map<String, String>` that lets user to create a map himselfe and inject it here.